File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 0
    hash_map = {}
    for name, category in clothes:
        if category in hash_map.keys():
            hash_map[category] += [name]
        else:
            hash_map[category] = [name]
    
    
    answer = 1
    for i, value in hash_map.items():
        answer *= (len(value) + 1)
    
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, HashSet<String>> sh = new HashMap<>();
        
        for (String[] cloth : clothes) {
            if (!sh.containsKey(cloth[1])) {
                sh.put(cloth[1], new HashSet<>());
            }
            sh.get(cloth[1]).add(cloth[0]);
        }
        
        for (String key : sh.keySet()) {
            answer *= (sh.get(key).size() + 1);
        }
        
        return answer - 1;
    }
}